### PR TITLE
Navigation Link: Match Editor Markup to Rendered

### LINKED ISSUE #28575: Navigation Block: Try contenteditable anchor (`a`) markup
Extracted from https://github.com/WordPress/gutenberg/pull/28265#issuecomment-768092543.

The closer the editor markup of a block matches that of the frontend, the easier it is to write CSS that just works. For the navigation block, the markup of a basic menu in the editor is this (lightly trimmed for clarity):

```
<ul class="wp-block-navigation__container block-editor-block-list__layout">
   <li class="wp-block-navigation-link block-editor-block-list__block wp-block has-link">
      <div class="wp-block-navigation-link__content">
         <div class="block-editor-rich-text__editable wp-block-navigation-link__label rich-text keep-placeholder-on-focus">Page link</div>
      </div>
   </li>
</ul>
```

Compare with the frontend:

```
<ul class="wp-block-navigation__container">
   <li class="wp-block-navigation-link">
      <a class="wp-block-navigation-link__content" href="http://localhost:8888/?page_id=265">
         <span class="wp-block-navigation-link__label">Page link</span>
      </a>
   </li>
</ul>
```

These are faaairly close on classes and markup, but the primary differentiator making styles differ is this one:

`<div class="wp-block-navigation-link__content">`

If that `div` could become an `a` instead, in the editor, the CSS to style both would be slightly simpler.

From a very superficial and maybe faulty investigation, [it seems like it should be possible to use a contenteditable anchor](https://codepen.io/joen/pen/KKgOrGK). So it seems worth exploring.

### PR DESCRIPTION
**Pull request by georgeh**
## Description

Closes #28575

Changes the element structure of the navigation link block in the editor to match the rendered output.

### Editor markup:
```html
<nav class="… wp-block-navigation" …>
	<ul class="wp-block-navigation__container …">
		<li class="… wp-block-navigation-link" …>
			<a class="wp-block-navigation-link__content">
				<div … class="block-editor-rich-text__editable …" contenteditable="true">example.com</div>
			</a>
			…
		</li>
		…
	</ul>
</nav>
```
### Rendered markup:
```html
<nav class="wp-block-navigation">
	<ul class="wp-block-navigation__container">
		<li class="wp-block-navigation-link">
			<a class="wp-block-navigation-link__content" href="http://example.com">
				<span class="wp-block-navigation-link__label">example.com</span>
			</a>
		</li>
	</ul>
</nav>
```

## How has this been tested?

Manually tested on desktop Firefox

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation.

### CODE REVIEW

**Comment by gwwar** [comment]
@jasmussen is it okay that the richtext/textarea is a div instead of span?

**Comment by jasmussen** [comment]
> @jasmussen is it okay that the richtext/textarea is a div instead of span?

Let me give it a quick spin. I take it the span can't be editable?

**Comment by gwwar** [comment]
So I think I found a few existing accessibility issues while testing with voiceover. If we'd like to tackle this in separate issues, let me know and I can file new tickets.

For folks 👀 , also let me know if certain behavior is expected or if there's alternative keyboard movement controls. (I am aware of keyboard shortcuts, though any richtext or text areas will capture keyboard input).

1. From the placeholder navigation menu, press <kbd>enter</kbd> on the "start empty" element.

Navigation doesn't appear to be selected, I need to <kbd>tab</kbd> again.

https://user-images.githubusercontent.com/1270189/111501019-d698fe80-8701-11eb-89cf-c5c83717632f.mp4

2. After adding a link, pressing <kbd>enter</kbd> then <kbd>tab</kbd> opens publishing settings?

https://user-images.githubusercontent.com/1270189/111500336-1d3a2900-8701-11eb-9049-0fdc85dff52e.mp4

Other keyboard shortcuts for reference.
Screen Shot 2021-03-17 at 9 23 40 AM

**Comment by jasmussen** [comment]
From a quick test, this looks amazing. Before in the editor:

Screenshot 2021-03-17 at 17 29 51

After in the editor:

Screenshot 2021-03-17 at 17 32 21

The frontend looks the same:

Screenshot 2021-03-17 at 17 30 15

Note, the above is the page list and a custom menu item, and I'm testing using "empty theme": https://github.com/WordPress/theme-experiments. The custom menu item does not have an underline, but that appears to be due to this CSS from https://github.com/WordPress/gutenberg/pull/26059:

```
.wp-block-navigation:not([style*="text-decoration"]) {
	.wp-block-navigation-link__content {
		text-decoration: none;

		&:focus,
		&:active {
			text-decoration: none;
		}
	}
}
```

Testing my own theme, which styles the a with a box shadow, here's before:

Screenshot 2021-03-17 at 17 22 49

and here's after:

Screenshot 2021-03-17 at 17 26 48

And after is what it's supposed to look like:

Screenshot 2021-03-17 at 17 22 52

In other words, this is working, and the fact that it surfaces that text-decoration rule is an _example_ of why this one is so important to land, because otherwise the CSS we write won't be able to easily affect both the Page List block, custom menu items, and the frontend at the same time, needing 3 separate rulesets. This is being reduced to one ruleset, which is just 👌 

Kerry about the keyboard interaction, in some contexts we use tab to switch region or section, and arrowkeys to navigate items within. That may be what's going on here. But just for this PR specifically, I'd compare with what's in trunk. There's certainly still some things we need to get right here, but just important we know whether they're tied to this PR or not.

**Review by gwwar — APPROVED** [review]
Thanks @georgeh for looking into this, I think this should be okay to try. I retested on trunk, for keyboard navigation and the issues I found here were also present. I can file a few issues as a follow up.

**Comment by jasmussen** [comment]
🎉

**Comment by gwwar** [comment]
Here's the a11y issue I filed: https://github.com/WordPress/gutenberg/issues/29948
